src/stripe.android.ts

```diff
@@ -80,37 +87,65 @@ export class Stripe {
   }
 
   retrievePaymentIntent(clientSecret: string, cb: (error: Error, pm: StripePaymentIntent) => void): void {
-    const params = com.stripe.android.model.PaymentIntentParams.createRetrievePaymentIntentParams(clientSecret);
     try {
-      const pi = this._stripe.retrievePaymentIntentSynchronous(params, this._apiKey);
-      if (typeof cb === 'function') {
-        cb(null, StripePaymentIntent.fromNative(pi));
-      }
+      const pi = this._stripe.retrievePaymentIntentSynchronous(clientSecret);
+      cb(null, StripePaymentIntent.fromNative(pi));
     } catch (error) {
-      if (typeof cb === 'function') {
-        cb(new Error(error.localizedDescription), null);
-      }
+      cb(new Error(error.localizedDescription), null);
     }
   }
 
-  confirmPaymentIntent(pi: StripePaymentIntent, returnUrl: string, cb: (error: Error, pm: StripePaymentIntent) => void): void {
-    const params = com.stripe.android.model.PaymentIntentParams.createConfirmPaymentIntentWithPaymentMethodId(
-      pi.id, pi.clientSecret, returnUrl);
+  confirmSetupIntent(paymentMethodId: string, clientSecret: string, cb: (error: Error, pm: StripeSetupIntent) => void): void {
     try {
-      const pi = this._stripe.confirmPaymentIntentSynchronous(params, this._apiKey);
-      if (typeof cb === 'function') {
-        cb(null, StripePaymentIntent.fromNative(pi));
-      }
+      const activity = androidApp.foregroundActivity;
+
+      const resultCb = new com.stripe.android.ApiResultCallback<com.stripe.android.SetupIntentResult>({
+        onSuccess: (result: com.stripe.android.SetupIntentResult) => {
+          const setupIntent = result.getIntent();
+          cb(null, StripeSetupIntent.fromNative(setupIntent))
+        },
+        onError: (error: any) => {
+          cb(new Error(error.localizedDescription), null)
```

Comment:
It should be `error.getLocalizedMessage()` instead of `error.localizedDescription` I believe the others occurrences of `error.localizedDescription` also need updated